src/lib/optimizer/abstract_syntax_tree/aggregate_node.cpp

```diff
@@ -29,50 +29,69 @@ const std::vector<std::shared_ptr<Expression>>& AggregateNode::aggregate_express
 
 const std::vector<ColumnID>& AggregateNode::groupby_column_ids() const { return _groupby_column_ids; }
 
-std::string AggregateNode::description() const {
-  std::ostringstream s;
+std::string AggregateNode::description(DescriptionMode mode) const {
+  std::ostringstream desc;
 
-  s << "[Aggregate] ";
+  desc << "[Aggregate] ";
+  if (mode == DescriptionMode::MultiLine) {
+    desc << "\n";
+  }
 
   std::vector<std::string> verbose_column_names;
   if (left_child()) {
     verbose_column_names = left_child()->get_verbose_column_names();
   }
 
-  auto stream_aggregate = [&](const std::shared_ptr<Expression>& aggregate_expr) {
-    s << aggregate_expr->to_string(verbose_column_names);
+  for (size_t aggregate_idx = 0; aggregate_idx < _aggregate_expressions.size(); ++aggregate_idx) {
+    const auto& aggregate_expr = _aggregate_expressions[aggregate_idx];
 
+    desc << aggregate_expr->to_string(verbose_column_names);
     if (aggregate_expr->alias()) {
-      s << " AS \"" << (*aggregate_expr->alias()) << "\"";
+      desc << " AS \"" << (*aggregate_expr->alias()) << "\"";
     }
-  };
-
-  auto aggregates_it = _aggregate_expressions.begin();
-  if (aggregates_it != _aggregate_expressions.end()) {
-    stream_aggregate(*aggregates_it);
-    ++aggregates_it;
-  }
 
-  for (; aggregates_it != _aggregate_expressions.end(); ++aggregates_it) {
-    s << ", ";
-    stream_aggregate(*aggregates_it);
+    if (aggregate_idx + 1 < _aggregate_expressions.size()) {
+      if (mode == DescriptionMode::SingleLine) {
+        desc << ", ";
+      } else {
+        desc << "\n";
+      }
```

Comment:
How do you feel about using the ternary operator here? That should make everything shorter and since it is only about strings, it's not really unsafe.

`desc << (multiline ? "\n" : ", ");`

Reply:
Okay, thought your stance was "no ternary!!", but changed it.

Reply:
The cases where I don't like it is if it spans multiple lines or if it has any side effects.